From 3438b4bf619e97a5dd043417d9bca8535baeb2d0 Mon Sep 17 00:00:00 2001
From: Matt Ellis <matell@microsoft.com>
Date: Thu, 18 May 2017 18:35:17 -0700
Subject: [PATCH 1/5] Add projects using the new dotnet sdk

These projects are built using the dotnet tooling instead of the
current system MSBuild has. Long term, MSBuild wants to start using
the SDK as well (see
https://github.com/Microsoft/msbuild/issues/1399), so we'll get ahead
of the game for the projects we care about for the CLI, which makes
building using just dotnet easier (no more project.json's to worry
about, for example).
---
 NetCore.sln                                        |  95 +++++
 build/NetCore.props                                |  21 +
 src/Build/Microsoft.Build.NetCore.csproj           | 466 +++++++++++++++++++++
 .../Microsoft.Build.Framework.NetCore.csproj       |  80 ++++
 src/MSBuild/MSBuild.NetCore.csproj                 | 195 +++++++++
 .../Microsoft.Build.Tasks.Core.NetCore.csproj      | 262 ++++++++++++
 .../Microsoft.Build.Utilities.Core.NetCore.csproj  | 115 +++++
 7 files changed, 1234 insertions(+)
 create mode 100644 NetCore.sln
 create mode 100644 build/NetCore.props
 create mode 100644 src/Build/Microsoft.Build.NetCore.csproj
 create mode 100644 src/Framework/Microsoft.Build.Framework.NetCore.csproj
 create mode 100644 src/MSBuild/MSBuild.NetCore.csproj
 create mode 100644 src/Tasks/Microsoft.Build.Tasks.Core.NetCore.csproj
 create mode 100644 src/Utilities/Microsoft.Build.Utilities.Core.NetCore.csproj

diff --git a/NetCore.sln b/NetCore.sln
new file mode 100644
index 00000000..8da57549
--- /dev/null
+++ b/NetCore.sln
@@ -0,0 +1,95 @@
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio 15
+VisualStudioVersion = 15.0.26006.2
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "src", "src", "{4EB318F1-78D9-4C1F-BDF8-502D1F7E4619}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MSBuild.NetCore", "src\MSBuild\MSBuild.NetCore.csproj", "{19EA6D6C-82DF-413B-AE1F-927CC9C2ABB6}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Microsoft.Build.Framework.NetCore", "src\Framework\Microsoft.Build.Framework.NetCore.csproj", "{86072A09-57B5-4F72-8329-1887BA344C27}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Microsoft.Build.Utilities.Core.NetCore", "src\Utilities\Microsoft.Build.Utilities.Core.NetCore.csproj", "{29E44D02-7EC4-4242-9086-0356985263D7}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Microsoft.Build.NetCore", "src\Build\Microsoft.Build.NetCore.csproj", "{00421582-9D72-4798-8D68-9D851F2C637F}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Microsoft.Build.Tasks.Core.NetCore", "src\Tasks\Microsoft.Build.Tasks.Core.NetCore.csproj", "{EDC05A6F-80B4-40E4-9B6F-2114726ACD0E}"
+EndProject
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|Any CPU = Debug|Any CPU
+		Debug|x64 = Debug|x64
+		Debug|x86 = Debug|x86
+		Release|Any CPU = Release|Any CPU
+		Release|x64 = Release|x64
+		Release|x86 = Release|x86
+	EndGlobalSection
+	GlobalSection(ProjectConfigurationPlatforms) = postSolution
+		{19EA6D6C-82DF-413B-AE1F-927CC9C2ABB6}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{19EA6D6C-82DF-413B-AE1F-927CC9C2ABB6}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{19EA6D6C-82DF-413B-AE1F-927CC9C2ABB6}.Debug|x64.ActiveCfg = Debug|x64
+		{19EA6D6C-82DF-413B-AE1F-927CC9C2ABB6}.Debug|x64.Build.0 = Debug|x64
+		{19EA6D6C-82DF-413B-AE1F-927CC9C2ABB6}.Debug|x86.ActiveCfg = Debug|x86
+		{19EA6D6C-82DF-413B-AE1F-927CC9C2ABB6}.Debug|x86.Build.0 = Debug|x86
+		{19EA6D6C-82DF-413B-AE1F-927CC9C2ABB6}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{19EA6D6C-82DF-413B-AE1F-927CC9C2ABB6}.Release|Any CPU.Build.0 = Release|Any CPU
+		{19EA6D6C-82DF-413B-AE1F-927CC9C2ABB6}.Release|x64.ActiveCfg = Release|x64
+		{19EA6D6C-82DF-413B-AE1F-927CC9C2ABB6}.Release|x64.Build.0 = Release|x64
+		{19EA6D6C-82DF-413B-AE1F-927CC9C2ABB6}.Release|x86.ActiveCfg = Release|x86
+		{19EA6D6C-82DF-413B-AE1F-927CC9C2ABB6}.Release|x86.Build.0 = Release|x86
+		{86072A09-57B5-4F72-8329-1887BA344C27}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{86072A09-57B5-4F72-8329-1887BA344C27}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{86072A09-57B5-4F72-8329-1887BA344C27}.Debug|x64.ActiveCfg = Debug|x64
+		{86072A09-57B5-4F72-8329-1887BA344C27}.Debug|x64.Build.0 = Debug|x64
+		{86072A09-57B5-4F72-8329-1887BA344C27}.Debug|x86.ActiveCfg = Debug|x86
+		{86072A09-57B5-4F72-8329-1887BA344C27}.Debug|x86.Build.0 = Debug|x86
+		{86072A09-57B5-4F72-8329-1887BA344C27}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{86072A09-57B5-4F72-8329-1887BA344C27}.Release|Any CPU.Build.0 = Release|Any CPU
+		{86072A09-57B5-4F72-8329-1887BA344C27}.Release|x64.ActiveCfg = Release|x64
+		{86072A09-57B5-4F72-8329-1887BA344C27}.Release|x64.Build.0 = Release|x64
+		{86072A09-57B5-4F72-8329-1887BA344C27}.Release|x86.ActiveCfg = Release|x86
+		{86072A09-57B5-4F72-8329-1887BA344C27}.Release|x86.Build.0 = Release|x86
+		{29E44D02-7EC4-4242-9086-0356985263D7}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{29E44D02-7EC4-4242-9086-0356985263D7}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{29E44D02-7EC4-4242-9086-0356985263D7}.Debug|x64.ActiveCfg = Debug|x64
+		{29E44D02-7EC4-4242-9086-0356985263D7}.Debug|x64.Build.0 = Debug|x64
+		{29E44D02-7EC4-4242-9086-0356985263D7}.Debug|x86.ActiveCfg = Debug|x86
+		{29E44D02-7EC4-4242-9086-0356985263D7}.Debug|x86.Build.0 = Debug|x86
+		{29E44D02-7EC4-4242-9086-0356985263D7}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{29E44D02-7EC4-4242-9086-0356985263D7}.Release|Any CPU.Build.0 = Release|Any CPU
+		{29E44D02-7EC4-4242-9086-0356985263D7}.Release|x64.ActiveCfg = Release|x64
+		{29E44D02-7EC4-4242-9086-0356985263D7}.Release|x64.Build.0 = Release|x64
+		{29E44D02-7EC4-4242-9086-0356985263D7}.Release|x86.ActiveCfg = Release|x86
+		{29E44D02-7EC4-4242-9086-0356985263D7}.Release|x86.Build.0 = Release|x86
+		{00421582-9D72-4798-8D68-9D851F2C637F}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{00421582-9D72-4798-8D68-9D851F2C637F}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{00421582-9D72-4798-8D68-9D851F2C637F}.Debug|x64.ActiveCfg = Debug|x64
+		{00421582-9D72-4798-8D68-9D851F2C637F}.Debug|x64.Build.0 = Debug|x64
+		{00421582-9D72-4798-8D68-9D851F2C637F}.Debug|x86.ActiveCfg = Debug|x86
+		{00421582-9D72-4798-8D68-9D851F2C637F}.Debug|x86.Build.0 = Debug|x86
+		{00421582-9D72-4798-8D68-9D851F2C637F}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{00421582-9D72-4798-8D68-9D851F2C637F}.Release|Any CPU.Build.0 = Release|Any CPU
+		{00421582-9D72-4798-8D68-9D851F2C637F}.Release|x64.ActiveCfg = Release|x64
+		{00421582-9D72-4798-8D68-9D851F2C637F}.Release|x64.Build.0 = Release|x64
+		{00421582-9D72-4798-8D68-9D851F2C637F}.Release|x86.ActiveCfg = Release|x86
+		{00421582-9D72-4798-8D68-9D851F2C637F}.Release|x86.Build.0 = Release|x86
+		{EDC05A6F-80B4-40E4-9B6F-2114726ACD0E}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{EDC05A6F-80B4-40E4-9B6F-2114726ACD0E}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{EDC05A6F-80B4-40E4-9B6F-2114726ACD0E}.Debug|x64.ActiveCfg = Debug|x64
+		{EDC05A6F-80B4-40E4-9B6F-2114726ACD0E}.Debug|x64.Build.0 = Debug|x64
+		{EDC05A6F-80B4-40E4-9B6F-2114726ACD0E}.Debug|x86.ActiveCfg = Debug|x86
+		{EDC05A6F-80B4-40E4-9B6F-2114726ACD0E}.Debug|x86.Build.0 = Debug|x86
+		{EDC05A6F-80B4-40E4-9B6F-2114726ACD0E}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{EDC05A6F-80B4-40E4-9B6F-2114726ACD0E}.Release|Any CPU.Build.0 = Release|Any CPU
+		{EDC05A6F-80B4-40E4-9B6F-2114726ACD0E}.Release|x64.ActiveCfg = Release|x64
+		{EDC05A6F-80B4-40E4-9B6F-2114726ACD0E}.Release|x64.Build.0 = Release|x64
+		{EDC05A6F-80B4-40E4-9B6F-2114726ACD0E}.Release|x86.ActiveCfg = Release|x86
+		{EDC05A6F-80B4-40E4-9B6F-2114726ACD0E}.Release|x86.Build.0 = Release|x86
+	EndGlobalSection
+	GlobalSection(NestedProjects) = preSolution
+		{19EA6D6C-82DF-413B-AE1F-927CC9C2ABB6} = {4EB318F1-78D9-4C1F-BDF8-502D1F7E4619}
+		{86072A09-57B5-4F72-8329-1887BA344C27} = {4EB318F1-78D9-4C1F-BDF8-502D1F7E4619}
+		{29E44D02-7EC4-4242-9086-0356985263D7} = {4EB318F1-78D9-4C1F-BDF8-502D1F7E4619}
+		{00421582-9D72-4798-8D68-9D851F2C637F} = {4EB318F1-78D9-4C1F-BDF8-502D1F7E4619}
+		{EDC05A6F-80B4-40E4-9B6F-2114726ACD0E} = {4EB318F1-78D9-4C1F-BDF8-502D1F7E4619}
+	EndGlobalSection
+EndGlobal
diff --git a/build/NetCore.props b/build/NetCore.props
new file mode 100644
index 00000000..e0777f09
--- /dev/null
+++ b/build/NetCore.props
@@ -0,0 +1,21 @@
+<Project>
+  <PropertyGroup>
+    <DisableImplicitFrameworkReferences>true</DisableImplicitFrameworkReferences>
+    <EnableDefaultItems>false</EnableDefaultItems>
+    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+    <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)MSFT.snk</AssemblyOriginatorKeyFile>
+    <SignAssembly>true</SignAssembly>
+    <PublicSign>true</PublicSign>
+    <VersionPrefix>15.3.0</VersionPrefix>
+    <VersionSuffix>preview-000234-01</VersionSuffix>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <DefineConstants>$(DefineConstants);TRACE;STANDALONEBUILD</DefineConstants>
+    <DefineConstants>$(DefineConstants);RUNTIME_TYPE_NETCORE</DefineConstants>
+    <DefineConstants>$(DefineConstants);FEATURE_CULTUREINFO_SETTERS</DefineConstants>
+    <DefineConstants>$(DefineConstants);FEATURE_PROCESSSTARTINFO_ENVIRONMENT</DefineConstants>
+    <DefineConstants>$(DefineConstants);USE_MSBUILD_DLL_EXTN</DefineConstants>
+    <DefineConstants>$(DefineConstants);STATIC_VERSION_NUMBER</DefineConstants>
+  </PropertyGroup>
+</Project>
diff --git a/src/Build/Microsoft.Build.NetCore.csproj b/src/Build/Microsoft.Build.NetCore.csproj
new file mode 100644
index 00000000..7bd793a9
--- /dev/null
+++ b/src/Build/Microsoft.Build.NetCore.csproj
@@ -0,0 +1,466 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="../../build/NetCore.props" />
+
+  <PropertyGroup>
+    <TargetFramework>netstandard1.5</TargetFramework>
+    <AssemblyName>Microsoft.Build</AssemblyName>
+    <DefineConstants>$(DefineConstants);BUILD_ENGINE</DefineConstants>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <DisableImplicitFrameworkReferences>true</DisableImplicitFrameworkReferences>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="NETStandard.Library" Version="1.6.0" />
+    <PackageReference Include="System.Collections.Immutable" Version="1.2.0" />
+    <PackageReference Include="System.Collections.NonGeneric" Version="4.0.1" />
+    <PackageReference Include="System.Diagnostics.Contracts" Version="4.0.1" />
+    <PackageReference Include="System.Diagnostics.FileVersionInfo" Version="4.0.0" />
+    <PackageReference Include="System.Diagnostics.Process" Version="4.1.0" />
+    <PackageReference Include="System.Diagnostics.TraceSource" Version="4.0.0" />
+    <PackageReference Include="System.IO.Compression" Version="4.1.0" />
+    <PackageReference Include="System.IO.FileSystem" Version="4.0.1" />
+    <PackageReference Include="System.IO.Pipes" Version="4.0.0" />
+    <PackageReference Include="System.Linq.Parallel" Version="4.0.1" />
+    <PackageReference Include="System.Reflection.Metadata" Version="1.3.0" />
+    <PackageReference Include="System.Reflection.TypeExtensions" Version="4.1.0" />
+    <PackageReference Include="System.Runtime" Version="4.1.0" />
+    <PackageReference Include="System.Runtime.InteropServices.RuntimeInformation" Version="4.0.0" />
+    <PackageReference Include="System.Runtime.Loader" Version="4.0.0" />
+    <PackageReference Include="System.Runtime.Serialization.Primitives" Version="4.1.1" />
+    <PackageReference Include="System.Text.Encoding.CodePages" Version="4.0.1" />
+    <PackageReference Include="System.Threading.Tasks.Dataflow" Version="4.6.0.0" />
+    <PackageReference Include="System.Threading.Thread" Version="4.0.0" />
+    <PackageReference Include="System.Threading.ThreadPool" Version="4.0.10" />
+    <PackageReference Include="System.Xml.XmlDocument" Version="4.0.1" />
+    <PackageReference Include="System.Xml.XPath.XmlDocument" Version="4.0.1" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="../Framework/Microsoft.Build.Framework.NetCore.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Compile Include="..\Shared\Compat\Base64Encoder.cs" />
+    <Compile Include="..\Shared\Compat\BinHexEncoder.cs" />
+    <Compile Include="..\Shared\Compat\SecureStringHasher.cs" />
+    <Compile Include="..\Shared\Compat\TypeExtensions.cs" />
+    <Compile Include="..\Shared\Compat\ValidateNames.cs" />
+    <Compile Include="..\Shared\Compat\XmlCharType.cs" />
+    <Compile Include="..\Shared\Compat\XmlReservedNs.cs" />
+    <Compile Include="..\Shared\Compat\XmlTextEncoder.cs" />
+    <Compile Include="..\Shared\Compat\XmlTextWriter.cs" />
+    <Compile Include="..\Shared\Compat\SerializableAttribute.cs" />
+    <Compile Include="..\Shared\Compat\defaultbinder.cs" />
+    <Compile Include="..\Shared\Compat\PipeSecurity.cs" />
+    <Compile Include="..\Shared\Compat\SafeHandleZeroOrMinusOneIsInvalid.cs" />
+    <Compile Include="..\Shared\EncodingUtilities.cs" />
+    <Compile Include="..\Shared\EnvironmentUtilities.cs" />
+    <Compile Include="..\Shared\FileUtilities.GetFolderPath.cs" />
+    <Compile Include="..\Shared\BuildEnvironmentHelper.cs" />
+    <Compile Include="..\Shared\EncodingStringWriter.cs" />
+    <Compile Include="..\Shared\FxCopExclusions\Microsoft.Build.Shared.Suppressions.cs" />
+    <Compile Include="..\Shared\AssemblyNameComparer.cs" />
+    <Compile Include="..\Shared\AwaitExtensions.cs" />
+    <Compile Include="..\Shared\AssemblyNameReverseVersionComparer.cs" />
+    <Compile Include="..\Shared\CanonicalError.cs" />
+    <Compile Include="..\Shared\Constants.cs" />
+    <Compile Include="..\Shared\HybridDictionary.cs" />
+    <Compile Include="..\Shared\NGen.cs" />
+    <Compile Include="..\Shared\Pair.cs" />
+    <Compile Include="..\Shared\PropertyParser.cs" />
+    <Compile Include="..\Shared\ReadOnlyEmptyCollection.cs" />
+    <Compile Include="..\Shared\StringBuilderCache.cs" />
+    <Compile Include="..\Shared\Traits.cs" />
+    <Compile Include="..\Shared\BufferedReadStream.cs" />
+    <Compile Include="..\Shared\TaskHostConfiguration.cs" />
+    <Compile Include="..\Shared\TaskHostTaskCancelled.cs" />
+    <Compile Include="..\Shared\TaskHostTaskComplete.cs" />
+    <Compile Include="..\Shared\OutOfProcTaskHostTaskResult.cs" />
+    <Compile Include="..\Shared\TaskLoader.cs" />
+    <Compile Include="..\Shared\NodeEngineShutdownReason.cs" />
+    <Compile Include="..\Shared\IKeyed.cs" />
+    <Compile Include="..\Shared\INodeEndpoint.cs" />
+    <Compile Include="..\Shared\NodeEndpointOutOfProcBase.cs" />
+    <Compile Include="..\Shared\INodePacket.cs" />
+    <Compile Include="..\Shared\INodePacketFactory.cs" />
+    <Compile Include="..\Shared\INodePacketHandler.cs" />
+    <Compile Include="..\Shared\INodePacketTranslatable.cs" />
+    <Compile Include="..\Shared\INodePacketTranslator.cs" />
+    <Compile Include="..\Shared\NodePacketTranslator.cs" />
+    <Compile Include="..\Shared\LogMessagePacketBase.cs" />
+    <Compile Include="..\Shared\NodePacketFactory.cs" />
+    <Compile Include="..\Shared\NodeBuildComplete.cs" />
+    <Compile Include="..\Shared\NodeShutdown.cs" />
+    <Compile Include="..\Shared\TaskLoggingHelper.cs" />
+    <Compile Include="..\Shared\TaskLoggingHelperExtension.cs" />
+    <Compile Include="..\Shared\TaskParameter.cs" />
+    <Compile Include="..\Shared\TaskParameterTypeVerifier.cs" />
+    <Compile Include="..\Shared\CommunicationsUtilities.cs" />
+    <Compile Include="..\Shared\InterningBinaryReader.cs" />
+    <Compile Include="..\Shared\TaskEngineAssemblyResolver.cs" />
+    <Compile Include="..\Shared\ReuseableStringBuilder.cs" />
+    <Compile Include="..\Shared\ThreadPoolExtensions.cs" />
+    <Compile Include="..\Shared\VisualStudioLocationHelper.cs" />
+    <Compile Include="AssemblyInfo.cs" />
+    <Compile Include="BackEnd\BuildManager\BuildManager.cs" />
+    <Compile Include="BackEnd\BuildManager\BuildParameters.cs" />
+    <Compile Include="BackEnd\BuildManager\BuildSubmission.cs" />
+    <Compile Include="BackEnd\BuildManager\LegacyThreadingData.cs" />
+    <Compile Include="BackEnd\Components\BuildComponentFactoryCollection.cs" />
+    <Compile Include="BackEnd\Components\Caching\IRegisteredTaskObjectCache.cs" />
+    <Compile Include="..\Shared\RegisteredTaskObjectCacheBase.cs" />
+    <Compile Include="BackEnd\Components\Caching\RegisteredTaskObjectCache.cs" />
+    <Compile Include="BackEnd\Components\Logging\EvaluationLoggingContext.cs" />
+    <Compile Include="BackEnd\Components\Logging\BuildLoggingContext.cs" />
+    <Compile Include="BackEnd\Components\Logging\LoggingContext.cs" />
+    <Compile Include="BackEnd\Components\Logging\BuildEventArgTransportSink.cs" />
+    <Compile Include="BackEnd\Components\Logging\CentralForwardingLogger.cs" />
+    <Compile Include="BackEnd\Components\Logging\EventRedirectorToSink.cs" />
+    <Compile Include="BackEnd\Components\Logging\EventSourceSink.cs" />
+    <Compile Include="BackEnd\Components\Logging\ForwardingLoggerRecord.cs" />
+    <Compile Include="BackEnd\Components\Logging\ILoggingService.cs" />
+    <Compile Include="BackEnd\Components\Logging\LoggingService.cs" />
+    <Compile Include="BackEnd\Components\Logging\LoggingServiceFactory.cs" />
+    <Compile Include="BackEnd\Components\Logging\LoggingServiceLogMethods.cs" />
+    <Compile Include="BackEnd\Components\Logging\NodeLoggingContext.cs" />
+    <Compile Include="BackEnd\Components\Logging\ProjectLoggingContext.cs" />
+    <Compile Include="BackEnd\Components\Logging\TargetLoggingContext.cs" />
+    <Compile Include="BackEnd\Components\Logging\TaskLoggingContext.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\IntrinsicTasks\CallTarget.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\IntrinsicTasks\IntrinsicTaskFactory.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\IntrinsicTasks\ItemGroupLoggingHelper.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\IntrinsicTasks\MSBuild.cs" />
+    <Compile Include="BackEnd\Components\Scheduler\SchedulableRequest.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\FullTracking.cs" />
+    <Compile Include="BackEnd\Components\Scheduler\SchedulingData.cs" />
+    <Compile Include="BackEnd\Components\Scheduler\Scheduler.cs" />
+    <Compile Include="BackEnd\Components\Scheduler\SchedulerCircularDependencyException.cs" />
+    <Compile Include="BackEnd\Components\Scheduler\ScheduleTimeRecord.cs" />
+    <Compile Include="BackEnd\Components\Scheduler\SchedulingPlan.cs" />
+    <Compile Include="BackEnd\Components\SdkResolution\DefaultSdkResolver.cs" />
+    <Compile Include="BackEnd\Components\SdkResolution\SdkResolution.cs" />
+    <Compile Include="BackEnd\Components\SdkResolution\SdkResolverLoader.cs" />
+    <Compile Include="BackEnd\Shared\ConfigurationMetadata.cs" />
+    <Compile Include="Collections\ConcurrentQueueExtensions.cs" />
+    <Compile Include="Collections\ConcurrentStackExtensions.cs" />
+    <Compile Include="Collections\LookasideStringInterner.cs" />
+    <Compile Include="Collections\RetrievableEntryHashSet\HashSet.cs" />
+    <Compile Include="Collections\RetrievableEntryHashSet\BitHelper.cs" />
+    <Compile Include="Collections\RetrievableEntryHashSet\HashSetDebugView.cs" />
+    <Compile Include="Collections\RetrievableEntryHashSet\HashHelpers.cs" />
+    <Compile Include="Construction\ImplicitImportLocation.cs" />
+    <Compile Include="Construction\ProjectSdkElement.cs" />
+    <Compile Include="Definition\NewProjectFileOptions.cs" />
+    <Compile Include="Definition\ProjectCollectionChangedEventArgs.cs" />
+    <Compile Include="Definition\ProjectImportPathMatch.cs" />
+    <Compile Include="Definition\ProjectLoadSettings.cs" />
+    <Compile Include="Definition\ToolsetLocalReader.cs" />
+    <Compile Include="Evaluation\ItemSpec.cs" />
+    <Compile Include="Evaluation\ItemsAndMetadataPair.cs" />
+    <Compile Include="Evaluation\LazyItemEvaluator.cs" />
+    <Compile Include="Evaluation\LazyItemEvaluator.IItemOperation.cs" />
+    <Compile Include="Evaluation\LazyItemEvaluator.UpdateOperation.cs" />
+    <Compile Include="Evaluation\LazyItemEvaluator.IncludeOperation.cs" />
+    <Compile Include="Evaluation\LazyItemEvaluator.ItemFactoryWrapper.cs" />
+    <Compile Include="Evaluation\LazyItemEvaluator.RemoveOperation.cs" />
+    <Compile Include="Evaluation\MetadataReference.cs" />
+    <Compile Include="Evaluation\ProjectChangedEventArgs.cs" />
+    <Compile Include="Evaluation\ProjectXmlChangedEventArgs.cs" />
+    <Compile Include="Construction\Solution\SolutionProjectGenerator.cs" />
+    <Compile Include="Evaluation\IToolsetProvider.cs" />
+    <Compile Include="BackEnd\Components\Caching\ResultsCacheResponse.cs" />
+    <Compile Include="BackEnd\Components\Communications\NodeEndpointOutOfProc.cs" />
+    <Compile Include="BackEnd\Components\Communications\NodeManager.cs" />
+    <Compile Include="BackEnd\Components\Communications\TaskHostNodeManager.cs" />
+    <Compile Include="BackEnd\Components\Communications\LogMessagePacket.cs" />
+    <Compile Include="BackEnd\Components\Communications\NodeFailedToLaunchException.cs" />
+    <Compile Include="BackEnd\Components\BuildRequestEngine\BuildRequestConfigurationResponse.cs" />
+    <Compile Include="BackEnd\Components\BuildRequestEngine\BuildRequestEngine.cs" />
+    <Compile Include="BackEnd\Components\BuildRequestEngine\BuildRequestEntry.cs" />
+    <Compile Include="BackEnd\Components\BuildRequestEngine\FullyQualifiedBuildRequest.cs" />
+    <Compile Include="BackEnd\Components\Caching\IConfigCache.cs" />
+    <Compile Include="BackEnd\Components\Caching\ResultsCache.cs" />
+    <Compile Include="BackEnd\Components\Caching\ConfigCache.cs" />
+    <Compile Include="BackEnd\Components\Caching\IPropertyCache.cs" />
+    <Compile Include="BackEnd\Components\Caching\IResultsCache.cs" />
+    <Compile Include="BackEnd\Components\Communications\NodePacketTranslatorExtensions.cs" />
+    <Compile Include="BackEnd\Components\Communications\NodeProviderOutOfProc.cs" />
+    <Compile Include="BackEnd\Components\Communications\NodeProviderOutOfProcBase.cs" />
+    <Compile Include="BackEnd\Components\Communications\NodeProviderOutOfProcTaskHost.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\BatchingEngine.cs" />
+    <Compile Include="BackEnd\BuildManager\BuildRequestData.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\IntrinsicTask.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\IntrinsicTasks\ItemGroupIntrinsicTask.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\IntrinsicTasks\PropertyGroupIntrinsicTask.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\ITargetBuilderCallback.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\TargetSpecification.cs" />
+    <Compile Include="BackEnd\Node\InProcNode.cs" />
+    <Compile Include="BackEnd\Node\NodeConfiguration.cs" />
+    <Compile Include="BackEnd\Node\OutOfProcNode.cs" />
+    <Compile Include="BackEnd\Node\NativeMethods.cs" />
+    <Compile Include="BackEnd\Shared\BuildAbortedException.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\IRequestBuilder.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\IRequestBuilderCallback.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\ITargetBuilder.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\ITaskBuilder.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\ItemBucket.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\Lookup.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\RequestBuilder.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\TargetBuilder.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\TargetEntry.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\TargetUpToDateChecker.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\TaskBuilder.cs" />
+    <Compile Include="BackEnd\Components\RequestBuilder\TaskHost.cs" />
+    <Compile Include="BackEnd\Shared\BuildRequest.cs" />
+    <Compile Include="BackEnd\Shared\BuildRequestBlocker.cs" />
+    <Compile Include="BackEnd\Shared\BuildRequestConfiguration.cs" />
+    <Compile Include="BackEnd\Shared\BuildResult.cs" />
+    <Compile Include="BackEnd\Shared\CircularDependencyException.cs" />
+    <Compile Include="BackEnd\Shared\BuildRequestUnblocker.cs" />
+    <Compile Include="BackEnd\Shared\IBuildResults.cs" />
+    <Compile Include="BackEnd\Shared\ITargetResult.cs" />
+    <Compile Include="BackEnd\Shared\TargetResult.cs" />
+    <Compile Include="BackEnd\Shared\WorkUnitResult.cs" />
+    <Compile Include="BackEnd\Components\BuildRequestEngine\IBuildRequestEngine.cs" />
+    <Compile Include="BackEnd\Components\Communications\INodeManager.cs" />
+    <Compile Include="BackEnd\Components\Communications\INodeProvider.cs" />
+    <Compile Include="BackEnd\Components\Communications\NodeEndpointInProc.cs" />
+    <Compile Include="BackEnd\Components\Communications\NodeInfo.cs" />
+    <Compile Include="BackEnd\Components\Communications\NodeProviderInProc.cs" />
+    <Compile Include="BackEnd\Components\IBuildComponent.cs" />
+    <Compile Include="BackEnd\Components\IBuildComponentHost.cs" />
+    <Compile Include="BackEnd\Components\Scheduler\IScheduler.cs" />
+    <Compile Include="BackEnd\Components\Scheduler\ScheduleResponse.cs" />
+    <Compile Include="BackEnd\Node\INode.cs" />
+    <Compile Include="BackEnd\TaskExecutionHost\AddInParts\ITaskExecutionHost.cs" />
+    <Compile Include="BackEnd\TaskExecutionHost\TaskExecutionHost.cs" />
+    <Compile Include="..\Shared\CollectionHelpers.cs" />
+    <Compile Include="Collections\ConvertingEnumerable.cs" />
+    <Compile Include="Collections\CopyOnReadEnumerable.cs" />
+    <Compile Include="..\Shared\OpportunisticIntern.cs" />
+    <Compile Include="..\Shared\CopyOnWriteDictionary.cs" />
+    <Compile Include="Collections\CopyOnWritePropertyDictionary.cs" />
+    <Compile Include="Collections\IDeepCloneable.cs" />
+    <Compile Include="..\Shared\IConstrainedEqualityComparer.cs" />
+    <Compile Include="..\Shared\MSBuildNameIgnoreCaseComparer.cs" />
+    <Compile Include="Collections\FilteringEnumerable.cs" />
+    <Compile Include="Collections\HashTableUtility.cs" />
+    <Compile Include="Collections\ItemDictionary.cs" />
+    <Compile Include="Collections\IImmutable.cs" />
+    <Compile Include="Collections\MultiDictionary.cs" />
+    <Compile Include="Collections\IValued.cs" />
+    <Compile Include="Collections\OrdinalKeyedComparer.cs" />
+    <Compile Include="Collections\OrdinalIgnoreCaseKeyedComparer.cs" />
+    <Compile Include="Collections\PropertyDictionary.cs" />
+    <Compile Include="..\Shared\ReadOnlyCollection.cs" />
+    <Compile Include="Collections\ReadOnlyConvertingCollection.cs" />
+    <Compile Include="Collections\ReadOnlyConvertingDictionary.cs" />
+    <Compile Include="Collections\Triple.cs" />
+    <Compile Include="Collections\WeakReference.cs" />
+    <Compile Include="Collections\WeakValueDictionary.cs" />
+    <Compile Include="Construction\ProjectElement.cs" />
+    <Compile Include="Construction\ProjectElementContainer.cs" />
+    <Compile Include="Construction\ProjectImportElement.cs" />
+    <Compile Include="Construction\ProjectImportGroupElement.cs" />
+    <Compile Include="Construction\ProjectItemDefinitionGroupElement.cs" />
+    <Compile Include="Construction\ProjectItemDefinitionElement.cs" />
+    <Compile Include="Construction\ProjectItemGroupElement.cs" />
+    <Compile Include="Construction\ProjectItemElement.cs" />
+    <Compile Include="Construction\ProjectMetadataElement.cs" />
+    <Compile Include="Construction\ProjectOnErrorElement.cs" />
+    <Compile Include="Construction\ProjectOtherwiseElement.cs" />
+    <Compile Include="Construction\ProjectOutputElement.cs" />
+    <Compile Include="Construction\ProjectExtensionsElement.cs" />
+    <Compile Include="Construction\ProjectPropertyGroupElement.cs" />
+    <Compile Include="Construction\ProjectPropertyElement.cs" />
+    <Compile Include="Construction\ProjectTargetElement.cs" />
+    <Compile Include="Construction\ProjectTaskElement.cs" />
+    <Compile Include="Construction\ProjectUsingTaskElement.cs" />
+    <Compile Include="Construction\ProjectRootElement.cs" />
+    <Compile Include="Construction\ProjectChooseElement.cs" />
+    <Compile Include="Construction\ProjectWhenElement.cs" />
+    <Compile Include="Construction\UsingTaskParameterGroupElement.cs" />
+    <Compile Include="Construction\ProjectUsingTaskParameterElement.cs" />
+    <Compile Include="Construction\ProjectUsingTaskBodyElement.cs" />
+    <Compile Include="Construction\Solution\SolutionConfigurationInSolution.cs" />
+    <Compile Include="Construction\Solution\ProjectConfigurationInSolution.cs" />
+    <Compile Include="Construction\Solution\ProjectInSolution.cs" />
+    <Compile Include="Construction\Solution\SolutionFile.cs" />
+    <Compile Include="Debugger\DebuggerLocalType.cs" />
+    <Compile Include="Debugger\DebuggerManager.cs" />
+    <Compile Include="Definition\BuiltInMetadata.cs" />
+    <Compile Include="Definition\ProjectCollection.cs" />
+    <Compile Include="Definition\Project.cs" />
+    <Compile Include="Definition\ProjectItem.cs" />
+    <Compile Include="Definition\ProjectItemDefinition.cs" />
+    <Compile Include="Definition\ProjectMetadata.cs" />
+    <Compile Include="Definition\ProjectProperty.cs" />
+    <Compile Include="Definition\ResolvedImport.cs" />
+    <Compile Include="Definition\SubToolset.cs" />
+    <Compile Include="Definition\Toolset.cs" />
+    <Compile Include="Definition\ToolsetPropertyDefinition.cs" />
+    <Compile Include="Definition\ToolsetReader.cs" />
+    <Compile Include="Definition\ToolsetRegistryReader.cs" />
+    <Compile Include="ElementLocation\ElementLocation.cs" />
+    <Compile Include="ElementLocation\RegistryLocation.cs" />
+    <Compile Include="ElementLocation\XmlAttributeWithLocation.cs" />
+    <Compile Include="ElementLocation\XmlDocumentWithLocation.cs" />
+    <Compile Include="ElementLocation\XmlElementWithLocation.cs" />
+    <Compile Include="ElementLocation\XmlNameTableThreadSafe.cs" />
+    <Compile Include="Errors\InternalLoggerException.cs" />
+    <Compile Include="Errors\InvalidProjectFileException.cs" />
+    <Compile Include="Errors\InvalidToolsetDefinitionException.cs" />
+    <Compile Include="Errors\RegistryException.cs" />
+    <Compile Include="Evaluation\ConditionEvaluator.cs" />
+    <Compile Include="Evaluation\Conditionals\AndExpressionNode.cs" />
+    <Compile Include="Evaluation\Conditionals\CharacterUtilities.cs" />
+    <Compile Include="Evaluation\Conditionals\EqualExpressionNode.cs" />
+    <Compile Include="Evaluation\Conditionals\FunctionCallExpressionNode.cs" />
+    <Compile Include="Evaluation\Conditionals\GenericExpressionNode.cs" />
+    <Compile Include="Evaluation\Conditionals\GreaterThanExpressionNode.cs" />
+    <Compile Include="Evaluation\Conditionals\GreaterThanOrEqualExpressionNode.cs" />
+    <Compile Include="Evaluation\Conditionals\LessThanExpressionNode.cs" />
+    <Compile Include="Evaluation\Conditionals\LessThanOrEqualExpressionNode.cs" />
+    <Compile Include="Evaluation\Conditionals\MultipleComparisonExpressionNode.cs" />
+    <Compile Include="Evaluation\Conditionals\NotEqualExpressionNode.cs" />
+    <Compile Include="Evaluation\Conditionals\NotExpressionNode.cs" />
+    <Compile Include="Evaluation\Conditionals\NumericComparisonExpressionNode.cs" />
+    <Compile Include="Evaluation\Conditionals\NumericExpressionNode.cs" />
+    <Compile Include="Evaluation\Conditionals\OperandExpressionNode.cs" />
+    <Compile Include="Evaluation\Conditionals\OperatorExpressionNode.cs" />
+    <Compile Include="Evaluation\Conditionals\OrExpressionNode.cs" />
+    <Compile Include="Evaluation\Conditionals\Parser.cs" />
+    <Compile Include="Evaluation\Conditionals\Scanner.cs" />
+    <Compile Include="Evaluation\Conditionals\StringExpressionNode.cs" />
+    <Compile Include="Evaluation\Conditionals\Token.cs" />
+    <Compile Include="Evaluation\EvaluatorMetadataTable.cs" />
+    <Compile Include="Evaluation\IEvaluatorData.cs" />
+    <Compile Include="Evaluation\IItem.cs" />
+    <Compile Include="Evaluation\IItemDefinition.cs" />
+    <Compile Include="Evaluation\IItemFactory.cs" />
+    <Compile Include="Evaluation\Conditionals\IItem.cs" />
+    <Compile Include="Evaluation\IItemProvider.cs" />
+    <Compile Include="Evaluation\IMetadataTable.cs" />
+    <Compile Include="Evaluation\IntrinsicFunctions.cs" />
+    <Compile Include="Evaluation\IMetadatum.cs" />
+    <Compile Include="Evaluation\IProjectMetadataParent.cs" />
+    <Compile Include="Evaluation\IProperty.cs" />
+    <Compile Include="Evaluation\IPropertyProvider.cs" />
+    <Compile Include="Evaluation\Preprocessor.cs" />
+    <Compile Include="Evaluation\ProjectParser.cs" />
+    <Compile Include="Evaluation\ProjectRootElementCache.cs" />
+    <Compile Include="Evaluation\ProjectStringCache.cs" />
+    <Compile Include="Evaluation\StringMetadataTable.cs" />
+    <Compile Include="Evaluation\ExpressionShredder.cs" />
+    <Compile Include="Evaluation\Evaluator.cs" />
+    <Compile Include="Evaluation\Expander.cs" />
+    <Compile Include="Evaluation\SemiColonTokenizer.cs" />
+    <Compile Include="Evaluation\ToolsetProvider.cs" />
+    <Compile Include="Globbing\CompositeGlob.cs" />
+    <Compile Include="Globbing\Extensions\MSBuildGlobExtensions.cs" />
+    <Compile Include="Globbing\Visitor\GlobVisitor.cs" />
+    <Compile Include="Globbing\MSBuildGlobWithGaps.cs" />
+    <Compile Include="Globbing\MSBuildGlob.cs" />
+    <Compile Include="Globbing\IMSBuildGlob.cs" />
+    <Compile Include="Globbing\Visitor\ParsedGlobCollector.cs" />
+    <Compile Include="Instance\ReflectableTaskPropertyInfo.cs" />
+    <Compile Include="Instance\HostServices.cs" />
+    <Compile Include="Instance\ProjectTargetInstanceChild.cs" />
+    <Compile Include="Instance\ProjectTaskInstanceChild.cs" />
+    <Compile Include="Instance\ProjectInstance.cs" />
+    <Compile Include="Instance\ProjectItemDefinitionInstance.cs" />
+    <Compile Include="Instance\ProjectItemGroupTaskInstance.cs" />
+    <Compile Include="Instance\ProjectItemGroupTaskItemInstance.cs" />
+    <Compile Include="Instance\ProjectItemGroupTaskMetadataInstance.cs" />
+    <Compile Include="Instance\ProjectItemInstance.cs" />
+    <Compile Include="Instance\ProjectMetadataInstance.cs" />
+    <Compile Include="Instance\ProjectOnErrorInstance.cs" />
+    <Compile Include="Instance\ProjectPropertyGroupTaskInstance.cs" />
+    <Compile Include="Instance\ProjectPropertyGroupTaskPropertyInstance.cs" />
+    <Compile Include="Instance\ProjectPropertyInstance.cs" />
+    <Compile Include="Instance\ProjectTargetInstance.cs" />
+    <Compile Include="Instance\ProjectTaskInstance.cs" />
+    <Compile Include="Instance\ProjectTaskOutputItemInstance.cs" />
+    <Compile Include="Instance\ProjectTaskOutputPropertyInstance.cs" />
+    <Compile Include="Instance\TaskFactories\AssemblyTaskFactory.cs" />
+    <Compile Include="Instance\TaskFactories\TaskHostTask.cs" />
+    <Compile Include="Instance\TaskFactoryLoggingHost.cs" />
+    <Compile Include="Instance\TaskFactoryWrapper.cs" />
+    <Compile Include="Instance\TaskRegistry.cs" />
+    <Compile Include="Evaluation\LazyItemEvaluator.LazyItemOperation.cs" />
+    <Compile Include="Logging\BaseConsoleLogger.cs" />
+    <Compile Include="Logging\BinaryLogger\BinaryLogger.cs" />
+    <Compile Include="Logging\BinaryLogger\BinaryLogRecordKind.cs" />
+    <Compile Include="Logging\BinaryLogger\BinaryLogReplayEventSource.cs" />
+    <Compile Include="Logging\BinaryLogger\BuildEventArgsDispatcher.cs" />
+    <Compile Include="Logging\BinaryLogger\BuildEventArgsFieldFlags.cs" />
+    <Compile Include="Logging\BinaryLogger\BuildEventArgsFields.cs" />
+    <Compile Include="Logging\BinaryLogger\BuildEventArgsReader.cs" />
+    <Compile Include="Logging\BinaryLogger\BuildEventArgsWriter.cs" />
+    <Compile Include="Logging\BinaryLogger\ProjectImportsCollector.cs" />
+    <Compile Include="Logging\ConsoleLogger.cs" />
+    <Compile Include="Logging\DistributedLoggers\DistributedFileLogger.cs" />
+    <Compile Include="Logging\DistributedLoggers\ConfigurableForwardingLogger.cs" />
+    <Compile Include="Logging\NullCentralLogger.cs" />
+    <Compile Include="Logging\LoggerDescription.cs" />
+    <Compile Include="Logging\ParallelLogger\ParallelLoggerHelpers.cs" />
+    <Compile Include="Logging\ParallelLogger\ParallelConsoleLogger.cs" />
+    <Compile Include="Logging\FileLogger.cs" />
+    <Compile Include="Logging\LogFormatter.cs" />
+    <Compile Include="Logging\SerialConsoleLogger.cs" />
+    <Compile Include="Resources\AssemblyResources.cs" />
+    <Compile Include="Resources\Constants.cs" />
+    <Compile Include="Utilities\EngineFileUtilities.cs" />
+    <Compile Include="Utilities\FileSpecMatcherTester.cs" />
+    <Compile Include="Utilities\RegistryKeyWrapper.cs" />
+    <Compile Include="Utilities\Utilities.cs" />
+    <Compile Include="Xml\ProjectXmlUtilities.cs" />
+    <Compile Include="Xml\ProjectXmlUtilities.XmlElementChildIterator.cs" />
+    <Compile Include="Xml\XmlReaderExtension.cs" />
+    <Compile Include="..\Shared\AssemblyLoadInfo.cs" />
+    <Compile Include="..\Shared\ReadOnlyEmptyDictionary.cs" />
+    <Compile Include="..\Shared\ReadOnlyEmptyList.cs" />
+    <Compile Include="..\Shared\AssemblyNameExtension.cs" />
+    <Compile Include="..\Shared\BuildEventFileInfo.cs" />
+    <Compile Include="..\Shared\ConversionUtilities.cs" />
+    <Compile Include="..\Shared\FileDelegates.cs" />
+    <Compile Include="..\Shared\ErrorUtilities.cs" />
+    <Compile Include="..\Shared\EscapingUtilities.cs" />
+    <Compile Include="..\Shared\VersionUtilities.cs" />
+    <Compile Include="..\Shared\EventArgsFormatting.cs" />
+    <Compile Include="..\Shared\ExceptionHandling.cs" />
+    <Compile Include="..\Shared\EscapingStringExtensions\EscapingStringExtensions.cs" />
+    <Compile Include="..\Shared\FileMatcher.cs" />
+    <Compile Include="..\Shared\FileUtilities.cs" />
+    <Compile Include="..\Shared\TempFileUtilities.cs" />
+    <Compile Include="..\Shared\Modifiers.cs" />
+    <Compile Include="..\Shared\FileUtilitiesRegex.cs" />
+    <Compile Include="..\Shared\FrameworkLocationHelper.cs" />
+    <Compile Include="..\Shared\IElementLocation.cs" />
+    <Compile Include="..\Shared\InternalErrorException.cs" />
+    <Compile Include="..\Shared\LoadedType.cs" />
+    <Compile Include="..\Shared\NativeMethodsShared.cs" />
+    <Compile Include="..\Shared\AssemblyUtilities.cs" />
+    <Compile Include="..\Shared\InprocTrackingNativeMethods.cs" />
+    <Compile Include="..\Shared\ProjectErrorUtilities.cs" />
+    <Compile Include="..\Shared\ProjectFileErrorUtilities.cs" />
+    <Compile Include="..\Shared\ProjectWriter.cs" />
+    <Compile Include="..\Shared\ResourceUtilities.cs" />
+    <Compile Include="..\Shared\Tracing.cs" />
+    <Compile Include="..\Shared\CoreCLRAssemblyLoader.cs" />
+    <Compile Include="..\Shared\TypeLoader.cs" />
+    <Compile Include="..\Shared\VisualStudioConstants.cs" />
+    <Compile Include="..\Shared\XMakeAttributes.cs" />
+    <Compile Include="..\Shared\XMakeElements.cs" />
+    <Compile Include="..\Shared\XmlUtilities.cs" />
+    <Compile Include="FxCopExclusions\Microsoft.Build.Suppressions.cs" />
+    <Compile Include="Evaluation\LazyItemEvaluator.EvaluatorData.cs" />
+
+    <!-- Resource Files -->
+    <EmbeddedResource Include="Resources\Strings.resx">
+      <LogicalName>$(AssemblyName).Strings.resources</LogicalName>
+      <SubType>Designer</SubType>
+    </EmbeddedResource>
+    <EmbeddedResource Include="..\Shared\Resources\Strings.shared.resx">
+      <SubType>Designer</SubType>
+      <LogicalName>$(AssemblyName).Strings.shared.resources</LogicalName>
diff --git a/src/Framework/Microsoft.Build.Framework.NetCore.csproj b/src/Framework/Microsoft.Build.Framework.NetCore.csproj
new file mode 100644
index 00000000..92f7aeae
--- /dev/null
+++ b/src/Framework/Microsoft.Build.Framework.NetCore.csproj
@@ -0,0 +1,80 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="../../build/NetCore.props" />
+
+  <PropertyGroup>
+    <TargetFramework>netstandard1.3</TargetFramework>
+    <AssemblyName>Microsoft.Build.Framework</AssemblyName>
+    <DisableImplicitFrameworkReferences>true</DisableImplicitFrameworkReferences>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="NETStandard.Library" Version="1.6.0" />
+    <PackageReference Include="System.Runtime.Serialization.Primitives" Version="4.1.1" />
+    <PackageReference Include="System.Text.Encoding.CodePages" Version="4.0.1" />
+    <PackageReference Include="System.Threading.Thread" Version="4.0.0" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Compile Include="..\Shared\Constants.cs" />
+    <Compile Include="AssemblyInfo.cs" />
+    <Compile Include="BuildEngineResult.cs" />
+    <Compile Include="BuildErrorEventArgs.cs" />
+    <Compile Include="BuildEventArgs.cs" />
+    <Compile Include="BuildFinishedEventArgs.cs" />
+    <Compile Include="BuildMessageEventArgs.cs" />
+    <Compile Include="BuildStartedEventArgs.cs" />
+    <Compile Include="BuildStatusEventArgs.cs" />
+    <Compile Include="BuildWarningEventArgs.cs" />
+    <Compile Include="CriticalBuildMessageEventArgs.cs" />
+    <Compile Include="CustomBuildEventArgs.cs" />
+    <Compile Include="ExternalProjectFinishedEventArgs.cs" />
+    <Compile Include="ExternalProjectStartedEventArgs.cs" />
+    <Compile Include="FxCopExclusions\Microsoft.Build.Framework.Suppressions.cs" />
+    <Compile Include="IBuildEngine3.cs" />
+    <Compile Include="IBuildEngine4.cs" />
+    <Compile Include="IBuildEngine5.cs" />
+    <Compile Include="IEventSource2.cs" />
+    <Compile Include="IGeneratedTask.cs" />
+    <Compile Include="LazyFormattedBuildEventArgs.cs" />
+    <Compile Include="IBuildEngine.cs" />
+    <Compile Include="IBuildEngine2.cs" />
+    <Compile Include="IEventRedirector.cs" />
+    <Compile Include="IEventSource.cs" />
+    <Compile Include="IForwardingLogger.cs" />
+    <Compile Include="ILogger.cs" />
+    <Compile Include="INodeLogger.cs" />
+    <Compile Include="ITask.cs" />
+    <Compile Include="ICancelableTask.cs" />
+    <Compile Include="ITaskFactory.cs" />
+    <Compile Include="ITaskFactory2.cs" />
+    <Compile Include="ITaskHost.cs" />
+    <Compile Include="ITaskItem.cs" />
+    <Compile Include="ITaskItem2.cs" />
+    <Compile Include="LoadInSeparateAppDomainAttribute.cs" />
+    <Compile Include="LoggerException.cs" />
+    <Compile Include="BuildEventContext.cs" />
+    <Compile Include="OutputAttribute.cs" />
+    <Compile Include="ProjectEvaluationFinishedEventArgs.cs" />
+    <Compile Include="ProjectEvaluationStartedEventArgs.cs" />
+    <Compile Include="ProjectImportedEventArgs.cs" />
+    <Compile Include="ProjectFinishedEventArgs.cs" />
+    <Compile Include="ProjectStartedEventArgs.cs" />
+    <Compile Include="Sdk\SdkReference.cs" />
+    <Compile Include="RequiredAttribute.cs" />
+    <Compile Include="RequiredRuntimeAttribute.cs" />
+    <Compile Include="RunInMTAAttribute.cs" />
+    <Compile Include="RunInSTAAttribute.cs" />
+    <Compile Include="Sdk\SdkLogger.cs" />
+    <Compile Include="Sdk\SdkResolver.cs" />
+    <Compile Include="Sdk\SdkResolverContext.cs" />
+    <Compile Include="Sdk\SdkResult.cs" />
+    <Compile Include="Sdk\SdkResultFactory.cs" />
+    <Compile Include="TargetFinishedEventArgs.cs" />
+    <Compile Include="TargetStartedEventArgs.cs" />
+    <Compile Include="TaskCommandLineEventArgs.cs" />
+    <Compile Include="TaskFinishedEventArgs.cs" />
+    <Compile Include="TaskPropertyInfo.cs" />
+    <Compile Include="TaskStartedEventArgs.cs" />
+    <Compile Include="TelemetryEventArgs.cs" />
+  </ItemGroup>
+</Project>
diff --git a/src/MSBuild/MSBuild.NetCore.csproj b/src/MSBuild/MSBuild.NetCore.csproj
new file mode 100644
index 00000000..ce87bfa4
--- /dev/null
+++ b/src/MSBuild/MSBuild.NetCore.csproj
@@ -0,0 +1,195 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="../../build/NetCore.props" />
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp1.0</TargetFramework>
+    <OutputType>Exe</OutputType>
+    <AssemblyName>MSBuild</AssemblyName>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <IncludeBuildOutput>false</IncludeBuildOutput>
+    <ContentTargetFolders>contentFiles</ContentTargetFolders>
+    <NoPackageAnalysis>true</NoPackageAnalysis>
+    <TargetMSBuildToolsVersion>15.0</TargetMSBuildToolsVersion>
+    <DisableImplicitFrameworkReferences>true</DisableImplicitFrameworkReferences>
+    <PackageId>Microsoft.Build.Runtime</PackageId>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NETCore.App" Version="1.0.1" />
+    <PackageReference Include="System.Collections.NonGeneric" Version="4.0.1" />
+    <PackageReference Include="System.Diagnostics.FileVersionInfo" Version="4.0.0" />
+    <PackageReference Include="System.Diagnostics.Process" Version="4.1.0" />
+    <PackageReference Include="System.Diagnostics.TraceSource" Version="4.0.0" />
+    <PackageReference Include="System.IO.Pipes" Version="4.0.0" />
+    <PackageReference Include="System.Reflection.Metadata" Version="1.3.0" />
+    <PackageReference Include="System.Reflection.TypeExtensions" Version="4.1.0" />
+    <PackageReference Include="System.Runtime.Loader" Version="4.0.0" />
+    <PackageReference Include="System.Runtime.InteropServices.RuntimeInformation" Version="4.0.0" />
+    <PackageReference Include="System.Threading.Thread" Version="4.0.0" />
+    <PackageReference Include="System.Threading.ThreadPool" Version="4.0.10" />
+    <PackageReference Include="System.Text.Encoding.CodePages" Version="4.0.1" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Projectreference Include="..\Build\Microsoft.Build.NetCore.csproj" />
+    <ProjectReference Include="..\Framework\Microsoft.Build.Framework.NetCore.csproj" />
+    <ProjectReference Include="..\Tasks\Microsoft.Build.Tasks.Core.NetCore.csproj" />
+    <ProjectReference Include="..\Utilities\Microsoft.Build.Utilities.Core.NetCore.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Compile Include="..\Shared\Compat\SerializableAttribute.cs" />
+    <Compile Include="..\Shared\Compat\SafeHandleZeroOrMinusOneIsInvalid.cs" />
+    <Compile Include="..\Shared\BuildEnvironmentHelper.cs" />
+    <Compile Include="..\Shared\FxCopExclusions\Microsoft.Build.Shared.Suppressions.cs" />
+    <Compile Include="..\Shared\AssemblyNameComparer.cs" />
+    <Compile Include="..\Shared\CollectionHelpers.cs" />
+    <Compile Include="..\Shared\Constants.cs" />
+    <Compile Include="..\Shared\FileDelegates.cs" />
+    <Compile Include="..\Shared\FileUtilities.cs" />
+    <Compile Include="..\Shared\FileUtilities.GetFolderPath.cs" />
+    <Compile Include="..\Shared\FileUtilitiesRegex.cs" />
+    <Compile Include="..\Shared\QuotingUtilities.cs" />
+    <Compile Include="..\Shared\RegisteredTaskObjectCacheBase.cs" />
+    <Compile Include="..\Shared\ResourceUtilities.cs" />
+    <Compile Include="..\Shared\EscapingUtilities.cs" />
+    <Compile Include="..\Shared\ErrorUtilities.cs" />
+    <Compile Include="..\Shared\Modifiers.cs" />
+    <Compile Include="..\Shared\BufferedReadStream.cs" />
+    <Compile Include="..\Shared\CopyOnWriteDictionary.cs" />
+    <Compile Include="..\Shared\HybridDictionary.cs" />
+    <Compile Include="..\Shared\IKeyed.cs" />
+    <Compile Include="..\Shared\IConstrainedEqualityComparer.cs" />
+    <Compile Include="..\Shared\MSBuildNameIgnoreCaseComparer.cs" />
+    <Compile Include="..\Shared\ReadOnlyEmptyCollection.cs" />
+    <Compile Include="..\Shared\ReadOnlyEmptyDictionary.cs" />
+    <Compile Include="..\Shared\ReadOnlyEmptyList.cs" />
+    <Compile Include="..\Shared\NativeMethodsShared.cs" />
+    <Compile Include="..\Shared\AssemblyUtilities.cs" />
+    <Compile Include="..\Shared\ExceptionHandling.cs" />
+    <Compile Include="..\Shared\InternalErrorException.cs" />
+    <Compile Include="..\Shared\EventArgsFormatting.cs" />
+    <Compile Include="..\Shared\TempFileUtilities.cs" />
+    <Compile Include="..\Shared\VisualStudioLocationHelper.cs" />
+    <Compile Include="..\Shared\XMakeAttributes.cs" />
+    <Compile Include="..\Shared\INodeEndpoint.cs" />
+    <Compile Include="..\Shared\NodeEndpointOutOfProcBase.cs" />
+    <Compile Include="..\Shared\INodePacket.cs" />
+    <Compile Include="..\Shared\LogMessagePacketBase.cs" />
+    <Compile Include="..\Shared\INodePacketFactory.cs" />
+    <Compile Include="..\Shared\NodePacketFactory.cs" />
+    <Compile Include="..\Shared\INodePacketHandler.cs" />
+    <Compile Include="..\Shared\INodePacketTranslatable.cs" />
+    <Compile Include="..\Shared\INodePacketTranslator.cs" />
+    <Compile Include="..\Shared\NodePacketTranslator.cs" />
+    <Compile Include="..\Shared\CommunicationsUtilities.cs" />
+    <Compile Include="..\Shared\InterningBinaryReader.cs" />
+    <Compile Include="..\Shared\OpportunisticIntern.cs" />
+    <Compile Include="..\Shared\TaskHostConfiguration.cs" />
+    <Compile Include="..\Shared\TaskHostTaskComplete.cs" />
+    <Compile Include="..\Shared\OutOfProcTaskHostTaskResult.cs" />
+    <Compile Include="..\Shared\TaskHostTaskCancelled.cs" />
+    <Compile Include="..\Shared\TaskLoader.cs" />
+    <Compile Include="..\Shared\TypeLoader.cs" />
+    <Compile Include="..\Shared\LoadedType.cs" />
+    <Compile Include="..\Shared\AssemblyLoadInfo.cs" />
+    <Compile Include="..\Shared\NodeBuildComplete.cs" />
+    <Compile Include="..\Shared\NodeShutdown.cs" />
+    <Compile Include="..\Shared\TaskParameter.cs" />
+    <Compile Include="..\Shared\TaskParameterTypeVerifier.cs" />
+    <Compile Include="..\Shared\TaskEngineAssemblyResolver.cs" />
+    <Compile Include="..\Shared\Traits.cs" />
+    <Compile Include="..\Shared\AssemblyNameExtension.cs" />
+    <Compile Include="..\Shared\ThreadPoolExtensions.cs" />
+    <Compile Include="..\Shared\StringBuilderCache.cs" />
+    <Compile Include="AssemblyInfo.cs" />
+    <Compile Include="AssemblyResources.cs" />
+    <Compile Include="CommandLineSwitches.cs" />
+    <Compile Include="CommandLineSwitchException.cs" />
+    <Compile Include="..\Shared\CoreCLRAssemblyLoader.cs" />
+    <Compile Include="DistributedLoggerRecord.cs" />
+    <Compile Include="FxCopExclusions\MsBuild.Suppressions.cs" />
+    <Compile Include="InitializationException.cs" />
+    <Compile Include="NodeEndpointOutOfProcTaskHost.cs" />
+    <Compile Include="LogMessagePacket.cs" />
+    <Compile Include="ProjectSchemaValidationHandler.cs" />
+    <Compile Include="OutOfProcTaskHostNode.cs" />
+    <Compile Include="OutOfProcTaskAppDomainWrapperBase.cs" />
+    <Compile Include="OutOfProcTaskAppDomainWrapper.cs" />
+    <Compile Include="XMake.cs" />
+
+    <EmbeddedResource Include="Resources\Strings.resx">
+      <LogicalName>$(AssemblyName).Strings.resources</LogicalName>
+      <SubType>Designer</SubType>
+    </EmbeddedResource>
+    <EmbeddedResource Include="..\Shared\Resources\Strings.shared.resx">
+      <LogicalName>$(AssemblyName).Strings.shared.resources</LogicalName>
+      <SubType>Designer</SubType>
+    </EmbeddedResource>
+
+    <Content Include="MSBuild.runtimeconfig.json">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="$(OutputPath)\MSBuild.dll">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.Common.CrossTargeting.targets">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.Common.CurrentVersion.targets">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.Common.overridetasks">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.Common.props">
+      <PackagePath>contentFiles\any\netcoreapp1.0\$(TargetMSBuildToolsVersion)</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.Common.targets">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.Common.tasks">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.CSharp.CrossTargeting.targets">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.CSharp.CurrentVersion.targets">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.CSharp.targets">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.NETFramework.CurrentVersion.props">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.NETFramework.CurrentVersion.targets">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.NETFramework.props">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.NETFramework.targets">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.VisualBasic.CrossTargeting.targets">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.VisualBasic.CurrentVersion.targets">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.VisualBasic.targets">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.VisualStudioVersion.v11.Common.props">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.VisualStudioVersion.v12.Common.props">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+    <Content Include="..\Tasks\Microsoft.VisualStudioVersion.v14.Common.props">
+      <PackagePath>contentFiles\any\netcoreapp1.0</PackagePath>
+    </Content>
+
+  </ItemGroup>
+
diff --git a/src/Tasks/Microsoft.Build.Tasks.Core.NetCore.csproj b/src/Tasks/Microsoft.Build.Tasks.Core.NetCore.csproj
new file mode 100644
index 00000000..ae654931
--- /dev/null
+++ b/src/Tasks/Microsoft.Build.Tasks.Core.NetCore.csproj
@@ -0,0 +1,262 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="../../build/NetCore.props" />
+
+  <PropertyGroup>
+    <TargetFramework>netstandard1.3</TargetFramework>
+    <AssemblyName>Microsoft.Build.Tasks.Core</AssemblyName>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <DefineConstants>$(DefineConstants);MICROSOFT_BUILD_TASKS</DefineConstants>
+    <DisableImplicitFrameworkReferences>true</DisableImplicitFrameworkReferences>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="NETStandard.Library" Version="1.6.0" />
+    <PackageReference Include="System.Collections.Immutable" Version="1.2.0" />
+    <PackageReference Include="System.Collections.NonGeneric" Version="4.0.1" />
+    <PackageReference Include="System.Diagnostics.Process" Version="4.1.0" />
+    <PackageReference Include="System.Diagnostics.TraceSource" Version="4.0.0" />
+    <PackageReference Include="System.Linq.Parallel" Version="4.0.1" />
+    <PackageReference Include="System.Reflection.Metadata" Version="1.3.0" />
+    <PackageReference Include="System.Reflection.TypeExtensions" Version="4.1.0" />
+    <PackageReference Include="System.Resources.Writer" Version="4.0.0" />
+    <PackageReference Include="System.Runtime.Serialization.Primitives" Version="4.1.1" />
+    <PackageReference Include="System.Runtime.Serialization.Xml" Version="4.1.1" />
+    <PackageReference Include="System.Runtime.InteropServices.RuntimeInformation" Version="4.0.0" />
+    <PackageReference Include="System.Security.Cryptography.Algorithms" Version="4.2.0" />
+    <PackageReference Include="System.Text.Encoding.CodePages" Version="4.0.1" />
+    <PackageReference Include="System.Threading.Thread" Version="4.0.0" />
+    <PackageReference Include="System.Xml.XmlDocument" Version="4.0.1" />
+    <PackageReference Include="System.Xml.XPath" Version="4.0.1" />
+    <PackageReference Include="System.Xml.XPath.XmlDocument" Version="4.0.1" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="../Framework/Microsoft.Build.Framework.NetCore.csproj" />
+    <ProjectReference Include="../Utilities/Microsoft.Build.Utilities.Core.NetCore.csproj" />
+  </ItemGroup>
+
+
+  <ItemGroup>
+    <!-- Source Files -->
+    <Compile Include="..\Shared\Compat\SerializableAttribute.cs" />
+    <Compile Include="..\Shared\Compat\SafeHandleZeroOrMinusOneIsInvalid.cs" />
+    <Compile Include="..\Shared\EnvironmentUtilities.cs" />
+    <Compile Include="..\Shared\AssemblyFolders\AssemblyFoldersEx.cs" />
+    <Compile Include="..\Shared\AssemblyFolders\AssemblyFoldersFromConfig.cs" />
+    <Compile Include="..\Shared\AssemblyFolders\Serialization\AssemblyFolderCollection.cs" />
+    <Compile Include="..\Shared\AssemblyFolders\Serialization\AssemblyFolderItem.cs" />
+    <Compile Include="..\Shared\BuildEnvironmentHelper.cs" />
+    <Compile Include="..\Shared\FxCopExclusions\Microsoft.Build.Shared.Suppressions.cs" />
+    <Compile Include="..\Shared\AssemblyNameComparer.cs" />
+    <Compile Include="..\Shared\AssemblyNameReverseVersionComparer.cs" />
+    <Compile Include="..\Shared\CanonicalError.cs" />
+    <Compile Include="..\Shared\Constants.cs" />
+    <Compile Include="ConvertToAbsolutePath.cs" />
+    <Compile Include="..\Shared\CopyOnWriteDictionary.cs" />
+    <Compile Include="..\Shared\ExtensionFoldersRegistryKey.cs" />
+    <Compile Include="..\Shared\FileDelegates.cs" />
+    <Compile Include="..\Shared\HybridDictionary.cs" />
+    <Compile Include="..\Shared\NativeMethodsShared.cs" />
+    <Compile Include="..\Shared\AssemblyUtilities.cs" />
+    <Compile Include="..\Shared\NGen.cs" />
+    <Compile Include="..\Shared\OpportunisticIntern.cs" />
+    <Compile Include="..\Shared\PropertyParser.cs" />
+    <Compile Include="..\Shared\ReadOnlyEmptyCollection.cs" />
+    <Compile Include="..\Shared\RegistryDelegates.cs" />
+    <Compile Include="..\Shared\RegistryHelper.cs" />
+    <Compile Include="..\Shared\StringBuilderCache.cs" />
+    <Compile Include="..\Shared\StrongNameHelpers.cs" />
+    <Compile Include="..\Shared\TaskLoggingHelperExtension.cs" />
+    <Compile Include="..\Shared\TempFileUtilities.cs" />
+    <Compile Include="..\Shared\MetadataConversionUtilities.cs" />
+    <Compile Include="..\Shared\LanguageParser\StreamMappedString.cs" />
+    <Compile Include="..\Shared\ExceptionHandling.cs" />
+    <Compile Include="..\Shared\FileUtilities.cs" />
+    <Compile Include="..\Shared\FileUtilities.GetFolderPath.cs" />
+    <Compile Include="..\Shared\EscapingUtilities.cs" />
+    <Compile Include="..\Shared\FileMatcher.cs" />
+    <Compile Include="..\Shared\Modifiers.cs" />
+    <Compile Include="..\Shared\ReadOnlyCollection.cs" />
+    <Compile Include="..\Shared\ReadOnlyEmptyDictionary.cs" />
+    <Compile Include="..\Shared\ReadOnlyEmptyList.cs" />
+    <Compile Include="..\Shared\Tracing.cs" />
+    <Compile Include="..\Shared\Traits.cs" />
+    <Compile Include="..\Shared\VersionUtilities.cs" />
+    <Compile Include="..\Shared\VisualStudioConstants.cs" />
+    <Compile Include="..\Shared\VisualStudioLocationHelper.cs" />
+    <Compile Include="..\Shared\AssemblyNameExtension.cs" />
+    <Compile Include="..\Shared\EncodingUtilities.cs" />
+    <Compile Include="..\Shared\ErrorUtilities.cs" />
+    <Compile Include="..\Shared\ConversionUtilities.cs" />
+    <Compile Include="..\Shared\FileUtilitiesRegex.cs" />
+    <Compile Include="..\Shared\InternalErrorException.cs" />
+    <Compile Include="..\Shared\ResourceUtilities.cs" />
+    <Compile Include="..\Shared\LanguageParser\token.cs" />
+    <Compile Include="..\Shared\LanguageParser\tokenChar.cs" />
+    <Compile Include="..\Shared\LanguageParser\tokenCharReader.cs" />
+    <Compile Include="..\Shared\LanguageParser\tokenEnumerator.cs" />
+    <Compile Include="..\Shared\LanguageParser\CSharptokenCharReader.cs" />
+    <Compile Include="..\Shared\LanguageParser\CSharptokenEnumerator.cs" />
+    <Compile Include="..\Shared\LanguageParser\CSharptokenizer.cs" />
+    <Compile Include="..\Shared\LanguageParser\VisualBasictokenCharReader.cs" />
+    <Compile Include="..\Shared\LanguageParser\VisualBasictokenEnumerator.cs" />
+    <Compile Include="..\Shared\LanguageParser\VisualBasictokenizer.cs" />
+    <Compile Include="AppConfig\*.cs" />
+    <Compile Include="AssemblyDependency\AssemblyFoldersExResolver.cs" />
+    <Compile Include="AssemblyDependency\AssemblyFoldersFromConfig\AssemblyFoldersFromConfigCache.cs" />
+    <Compile Include="AssemblyDependency\AssemblyFoldersFromConfig\AssemblyFoldersFromConfigResolver.cs" />
+    <Compile Include="AssemblyDependency\AssemblyFoldersResolver.cs" />
+    <Compile Include="AssemblyDependency\AssemblyInformation.cs" />
+    <Compile Include="AssemblyDependency\AssemblyNameReference.cs" />
+    <Compile Include="AssemblyDependency\AssemblyNameReferenceAscendingVersionComparer.cs" />
+    <Compile Include="AssemblyDependency\AssemblyResolution.cs" />
+    <Compile Include="AssemblyDependency\AssemblyResolutionConstants.cs" />
+    <Compile Include="AssemblyDependency\BadImageReferenceException.cs" />
+    <Compile Include="AssemblyDependency\CandidateAssemblyFilesResolver.cs" />
+    <Compile Include="AssemblyDependency\ConflictLossReason.cs" />
+    <Compile Include="AssemblyDependency\CopyLocalState.cs" />
+    <Compile Include="AssemblyDependency\DependencyResolutionException.cs" />
+    <Compile Include="AssemblyDependency\DirectoryResolver.cs" />
+    <Compile Include="AssemblyDependency\DisposableBase.cs" />
+    <Compile Include="AssemblyDependency\FrameworkPathResolver.cs" />
+    <Compile Include="AssemblyDependency\GenerateBindingRedirects.cs" />
+    <Compile Include="AssemblyDependency\HintPathResolver.cs" />
+    <Compile Include="AssemblyDependency\InstalledAssemblies.cs" />
+    <Compile Include="AssemblyDependency\InvalidReferenceAssemblyNameException.cs" />
+    <Compile Include="AssemblyDependency\NoMatchReason.cs" />
+    <Compile Include="AssemblyDependency\RawFilenameResolver.cs" />
+    <Compile Include="AssemblyDependency\Reference.cs" />
+    <Compile Include="AssemblyDependency\ReferenceResolutionException.cs" />
+    <Compile Include="AssemblyDependency\ReferenceTable.cs" />
+    <Compile Include="AssemblyDependency\ResolutionSearchLocation.cs" />
+    <Compile Include="AssemblyDependency\Resolver.cs" />
+    <Compile Include="AssemblyDependency\ResolveAssemblyReference.cs" />
+    <Compile Include="AssemblyDependency\TaskItemSpecFilenameComparer.cs" />
+    <Compile Include="AssemblyDependency\UnificationReason.cs" />
+    <Compile Include="AssemblyDependency\UnificationVersion.cs" />
+    <Compile Include="AssemblyDependency\UnifiedAssemblyName.cs" />
+    <Compile Include="AssemblyDependency\WarnOrErrorOnTargetArchitectureMismatchBehavior.cs" />
+    <Compile Include="AssemblyFolder.cs" />
+    <Compile Include="AssemblyInfo.cs" />
+    <Compile Include="AssemblyRemapping.cs" />
+    <Compile Include="AssemblyResources.cs" />
+    <Compile Include="AssignLinkMetadata.cs" />
+    <Compile Include="AssignProjectConfiguration.cs" />
+    <Compile Include="AssignTargetPath.cs" />
+    <Compile Include="CallTarget.cs" />
+    <Compile Include="CombinePath.cs" />
+    <Compile Include="CommandLineBuilderExtension.cs" />
+    <Compile Include="BuildCacheDisposeWrapper.cs" />
+    <Compile Include="FileState.cs" />
+    <Compile Include="Copy.cs" />
+    <Compile Include="CreateCSharpManifestResourceName.cs" />
+    <Compile Include="CreateVisualBasicManifestResourceName.cs" />
+    <Compile Include="CreateItem.cs" />
+    <Compile Include="CreateManifestResourceName.cs" />
+    <Compile Include="CreateProperty.cs" />
+    <Compile Include="CSharpParserUtilities.cs" />
+    <Compile Include="Delegate.cs" />
+    <Compile Include="Delete.cs" />
+    <Compile Include="Error.cs" />
+    <Compile Include="Exec.cs" />
+    <Compile Include="FindAppConfigFile.cs" />
+    <Compile Include="GetFrameworkPath.cs" />
+    <Compile Include="GetReferenceAssemblyPaths.cs" />
+    <Compile Include="Hash.cs" />
+    <Compile Include="InstalledSDKResolver.cs" />
+    <Compile Include="ErrorFromResources.cs" />
+    <Compile Include="ExtractedClassName.cs" />
+    <Compile Include="FileIO\ReadLinesFromFile.cs" />
+    <Compile Include="FileIO\WriteLinesToFile.cs" />
+    <Compile Include="FindInList.cs" />
+    <Compile Include="FormatVersion.cs" />
+    <Compile Include="FxCopExclusions\Microsoft.Build.Tasks.Suppressions.cs" />
+    <Compile Include="GenerateResource.cs" />
+    <Compile Include="IAnalyzerHostObject.cs" />
+    <Compile Include="ICscHostObject.cs" />
+    <Compile Include="ICscHostObject2.cs" />
+    <Compile Include="ICscHostObject3.cs" />
+    <Compile Include="ICscHostObject4.cs" />
+    <Compile Include="IVbcHostObject.cs" />
+    <Compile Include="IVbcHostObject2.cs" />
+    <Compile Include="IVbcHostObject3.cs" />
+    <Compile Include="IVbcHostObject4.cs" />
+    <Compile Include="IVbcHostObject5.cs" />
+    <Compile Include="IVbcHostObjectFreeThreaded.cs" />
+    <Compile Include="InvalidParameterValueException.cs" />
+    <Compile Include="ListOperators\FindUnderPath.cs" />
+    <Compile Include="ListOperators\RemoveDuplicates.cs" />
+    <Compile Include="MakeDir.cs" />
+    <Compile Include="Message.cs" />
+    <Compile Include="Move.cs" />
+    <Compile Include="MSBuild.cs" />
+    <Compile Include="NativeMethods.cs" />
+    <Compile Include="ParserState.cs" />
+    <Compile Include="RedistList.cs" />
+    <Compile Include="RemoveDir.cs" />
+    <Compile Include="ResolveCodeAnalysisRuleSet.cs" />
+    <Compile Include="ResolveKeySource.cs" />
+    <Compile Include="ResolveProjectBase.cs" />
+    <Compile Include="TaskExtension.cs" />
+    <Compile Include="Telemetry.cs" />
+    <Compile Include="ToolTaskExtension.cs" />
+    <Compile Include="Touch.cs" />
+    <Compile Include="VisualBasicParserUtilities.cs" />
+    <Compile Include="Warning.cs" />
+    <Compile Include="AssignCulture.cs" />
+    <Compile Include="Culture.cs" />
+    <Compile Include="CultureInfoCache.cs" />
+    <Compile Include="WriteCodeFragment.cs" />
+    <Compile Include="XmlPeek.cs" />
+    <Compile Include="XmlPoke.cs" />
+    <Compile Include="CodeTaskFactory.cs" />
+    <Compile Include="XamlTaskFactory\XamlTaskFactory.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <EmbeddedResource Include="system.design\system.design.txt">
+      <Type>Resx</Type>
+      <WithCulture>false</WithCulture>
+      <ManifestResourceName>%(Filename)</ManifestResourceName>
+      <FullClassName>Microsoft.Build.Tasks.SR</FullClassName>
+      <LogicalName>System.Design.resources</LogicalName>
+    </EmbeddedResource>
+    <EmbeddedResource Include="Resources\Strings.resx">
+      <LogicalName>$(AssemblyName).Strings.resources</LogicalName>
+      <SubType>Designer</SubType>
+    </EmbeddedResource>
+    <EmbeddedResource Include="..\Shared\Resources\Strings.shared.resx">
+      <SubType>Designer</SubType>
+      <LogicalName>$(AssemblyName).Strings.shared.resources</LogicalName>
+    </EmbeddedResource>
+    <EmbeddedResource Include="ManifestUtil\Resources\Strings.ManifestUtilities.resx">
+      <LogicalName>$(AssemblyName).Strings.ManifestUtilities.resources</LogicalName>
+      <SubType>Designer</SubType>
+    </EmbeddedResource>
+    <!-- Other Source Files -->
+    <EmbeddedResource Include="ManifestUtil\manifest.xml">
+      <LogicalName>Microsoft.Build.Tasks.Deployment.ManifestUtilities.manifest.xml</LogicalName>
+    </EmbeddedResource>
+    <EmbeddedResource Include="ManifestUtil\merge.xsl">
+      <LogicalName>Microsoft.Build.Tasks.Deployment.ManifestUtilities.merge.xsl</LogicalName>
+    </EmbeddedResource>
+    <EmbeddedResource Include="ManifestUtil\read2.xsl">
+      <LogicalName>Microsoft.Build.Tasks.Deployment.ManifestUtilities.read2.xsl</LogicalName>
+    </EmbeddedResource>
+    <EmbeddedResource Include="ManifestUtil\trustinfo2.xsl">
+      <LogicalName>Microsoft.Build.Tasks.Deployment.ManifestUtilities.trustinfo2.xsl</LogicalName>
+    </EmbeddedResource>
+    <EmbeddedResource Include="ManifestUtil\write2.xsl">
+      <LogicalName>Microsoft.Build.Tasks.Deployment.ManifestUtilities.write2.xsl</LogicalName>
+      <SubType>Designer</SubType>
+    </EmbeddedResource>
+    <EmbeddedResource Include="ManifestUtil\write3.xsl">
+      <LogicalName>Microsoft.Build.Tasks.Deployment.ManifestUtilities.write3.xsl</LogicalName>
+      <SubType>Designer</SubType>
+    </EmbeddedResource>
+    <EmbeddedResource Include="BootstrapperUtil\xmltoconfig.xsl">
+      <LogicalName>Microsoft.Build.Tasks.Deployment.Bootstrapper.xmltoconfig.xsl</LogicalName>
+    </EmbeddedResource>
+  </ItemGroup>
+
+</Project>
diff --git a/src/Utilities/Microsoft.Build.Utilities.Core.NetCore.csproj b/src/Utilities/Microsoft.Build.Utilities.Core.NetCore.csproj
new file mode 100644
index 00000000..2a4fbb4c
--- /dev/null
+++ b/src/Utilities/Microsoft.Build.Utilities.Core.NetCore.csproj
@@ -0,0 +1,115 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="../../build/NetCore.props" />
+
+  <PropertyGroup>
+    <TargetFramework>netstandard1.3</TargetFramework>
+    <AssemblyName>Microsoft.Build.Utilities.Core</AssemblyName>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="NETStandard.Library" Version="1.6.0" />
+    <PackageReference Include="System.Collections.Immutable" Version="1.2.0" />
+    <PackageReference Include="System.Collections.NonGeneric" Version="4.0.1" />
+    <PackageReference Include="System.Diagnostics.Process" Version="4.1.0" />
+    <PackageReference Include="System.Diagnostics.TraceSource" Version="4.0.0" />
+    <PackageReference Include="System.Reflection.TypeExtensions" Version="4.1.0" />
+    <PackageReference Include="System.Runtime.Serialization.Primitives" Version="4.1.1" />
+    <PackageReference Include="System.Runtime.Serialization.Xml" Version="4.1.1" />
+    <PackageReference Include="System.Runtime.InteropServices.RuntimeInformation" Version="4.0.0" />
+    <PackageReference Include="System.Security.Cryptography.X509Certificates" Version="4.1.0" />
+    <PackageReference Include="System.Text.Encoding.CodePages" Version="4.0.1" />
+    <PackageReference Include="System.Threading.Thread" Version="4.0.0" />
+    <PackageReference Include="System.Xml.XmlDocument" Version="4.0.1" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="../Framework/Microsoft.Build.Framework.NetCore.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Compile Include="..\Shared\Compat\SafeHandleZeroOrMinusOneIsInvalid.cs" />
+    <Compile Include="..\Shared\AssemblyFolders\AssemblyFoldersEx.cs" />
+    <Compile Include="..\Shared\AssemblyFolders\AssemblyFoldersFromConfig.cs" />
+    <Compile Include="..\Shared\AssemblyFolders\Serialization\AssemblyFolderCollection.cs" />
+    <Compile Include="..\Shared\AssemblyFolders\Serialization\AssemblyFolderItem.cs" />
+    <Compile Include="..\Shared\EnvironmentUtilities.cs" />
+    <Compile Include="..\Shared\FileUtilities.GetFolderPath.cs" />
+    <Compile Include="..\Shared\BuildEnvironmentHelper.cs" />
+    <Compile Include="..\Shared\FxCopExclusions\Microsoft.Build.Shared.Suppressions.cs" />
+    <Compile Include="..\Shared\CanonicalError.cs" />
+    <Compile Include="..\Shared\Constants.cs" />
+    <Compile Include="..\Shared\ExtensionFoldersRegistryKey.cs" />
+    <Compile Include="..\Shared\FileDelegates.cs" />
+    <Compile Include="..\Shared\CopyOnWriteDictionary.cs" />
+    <Compile Include="..\Shared\EncodingUtilities.cs" />
+    <Compile Include="..\Shared\ErrorUtilities.cs" />
+    <Compile Include="..\Shared\EscapingUtilities.cs" />
+    <Compile Include="..\Shared\EventArgsFormatting.cs" />
+    <Compile Include="..\Shared\ExceptionHandling.cs" />
+    <Compile Include="..\Shared\EscapingStringExtensions\EscapingStringExtensions.cs" />
+    <Compile Include="..\Shared\FileUtilities.cs" />
+    <Compile Include="..\Shared\FileMatcher.cs" />
+    <Compile Include="..\Shared\FileUtilitiesRegex.cs" />
+    <Compile Include="..\Shared\FrameworkLocationHelper.cs" />
+    <Compile Include="..\Shared\HybridDictionary.cs" />
+    <Compile Include="..\Shared\InternalErrorException.cs" />
+    <Compile Include="..\Shared\IKeyed.cs" />
+    <Compile Include="..\Shared\IConstrainedEqualityComparer.cs" />
+    <Compile Include="..\Shared\MSBuildNameIgnoreCaseComparer.cs" />
+    <Compile Include="..\Shared\Modifiers.cs" />
+    <Compile Include="..\Shared\NativeMethodsShared.cs" />
+    <Compile Include="..\Shared\AssemblyUtilities.cs" />
+    <Compile Include="..\Shared\InprocTrackingNativeMethods.cs" />
+    <Compile Include="..\Shared\OpportunisticIntern.cs" />
+    <Compile Include="..\Shared\ReadOnlyEmptyCollection.cs" />
+    <Compile Include="..\Shared\ReadOnlyEmptyDictionary.cs" />
+    <Compile Include="..\Shared\ReadOnlyEmptyList.cs" />
+    <Compile Include="..\Shared\RegistryDelegates.cs" />
+    <Compile Include="..\Shared\RegistryHelper.cs" />
+    <Compile Include="..\Shared\ResourceUtilities.cs" />
+    <Compile Include="..\Shared\StringBuilderCache.cs" />
+    <Compile Include="..\Shared\TaskLoggingHelper.cs" />
+    <Compile Include="..\Shared\TempFileUtilities.cs" />
+    <Compile Include="..\Shared\Tracing.cs" />
+    <Compile Include="..\Shared\VersionUtilities.cs" />
+    <Compile Include="..\Shared\ToolsetElement.cs" />
+    <Compile Include="..\Shared\VisualStudioLocationHelper.cs" />
+    <Compile Include="..\Shared\Traits.cs" />
+    <Compile Include="ApiContract.cs" />
+    <Compile Include="AppDomainIsolatedTask.cs" />
+    <Compile Include="AssemblyFolders\AssemblyFoldersExInfo.cs" />
+    <Compile Include="AssemblyFolders\AssemblyFoldersFromConfigInfo.cs" />
+    <Compile Include="AssemblyInfo.cs" />
+    <Compile Include="AssemblyResources.cs" />
+    <Compile Include="CommandLineBuilder.cs" />
+    <Compile Include="FxCopExclusions\Microsoft.Build.Utilities.Suppressions.cs" />
+    <Compile Include="Logger.cs" />
+    <Compile Include="MuxLogger.cs" />
+    <Compile Include="ProcessorArchitecture.cs" />
+    <Compile Include="SDKType.cs" />
+    <Compile Include="TargetPlatformSDK.cs" />
+    <Compile Include="ExtensionSDK.cs" />
+    <Compile Include="SDKManifest.cs" />
+    <Compile Include="PlatformManifest.cs" />
+    <Compile Include="Task.cs" />
+    <Compile Include="TaskItem.cs" />
+    <Compile Include="ToolLocationHelper.cs" />
+    <Compile Include="ToolTask.cs" />
+    <Compile Include="ProcessExtensions.cs" />
+    <Compile Include="TrackedDependencies\CanonicalTrackedInputFiles.cs" />
+    <Compile Include="TrackedDependencies\DependencyTableCache.cs" />
+    <Compile Include="TrackedDependencies\FlatTrackingData.cs" />
+    <Compile Include="TrackedDependencies\FileTracker.cs" />
+    <Compile Include="TrackedDependencies\CanonicalTrackedOutputFiles.cs" />
+    <Compile Include="TrackedDependencies\TrackedDependencies.cs" />
+    <Compile Include="TrackedDependencies\CanonicalTrackedFilesHelper.cs" />
+    <EmbeddedResource Include="Resources\Strings.resx">
+      <LogicalName>$(AssemblyName).Strings.resources</LogicalName>
+      <SubType>Designer</SubType>
+    </EmbeddedResource>
+    <EmbeddedResource Include="..\Shared\Resources\Strings.shared.resx">
+      <Link>Resources\Strings.shared.resx</Link>
+      <SubType>Designer</SubType>
+      <LogicalName>$(AssemblyName).Strings.shared.resources</LogicalName>
+    </EmbeddedResource>
+  </ItemGroup>
+
+</Project>
-- 
2.16.1.windows.4

